binary_search_tree/tree.py

```diff
@@ -1,59 +1,124 @@
 class TreeNode:
-    def __init__(self, key, val = None):
+    def __init__(self, key, val=None):
         if val == None:
             val = key
 
         self.key = key
         self.value = val
         self.left = None
         self.right = None
-        
 
 
 class Tree:
     def __init__(self):
         self.root = None
 
-    # Time Complexity: 
-    # Space Complexity: 
-    def add(self, key, value = None):
-        pass
+    def add_helper(self, current_node, key, value):
+        if current_node == None:
+            return TreeNode(key, value)
+
+        if key <= current_node.key:
+            current_node.left = self.add_helper(current_node.left, key, value)
+        else:
+            current_node.right = self.add_helper(
+                current_node.right, key, value)
+        return current_node
+
+    # Time Complexity: O logn
+    # Space Complexity: O n
+    def add(self, key, value=None):
```

Comment:
👍 However the time/space complexity is O(log n) if the tree is balanced O(n) if not.